File: python/demos/demo_lazy_theta_star.py
#!/usr/bin/env python3
"""Lazy Theta* demo."""

from __future__ import annotations

from demo_common import run_discrete

# Imported from the concrete module rather than the package top level so the demo
# runs before the flat re-export is added to navigation.global_planning.__init__.
from navigation.global_planning.search.lazy_theta_star import LazyThetaStar

if __name__ == "__main__":
    run_discrete("lazy_theta_star", LazyThetaStar)

File: cpp/demos/demo_lazy_theta_star.cpp
#include "demo_common.hpp"
#include "navigation/global_planning/search/lazy_theta_star.hpp"

int main(int argc, char** argv) {
  try {
    demo::Args args = demo::parse_args(argc, argv);
    auto params = navigation::core::ParamSet::from_yaml(args.params);
    navigation::global_planning::LazyThetaStarPlanner planner(params);
    return demo::run_discrete(args, params, planner);
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
